pallets/identity/src/lib.rs

```diff
@@ -967,7 +967,11 @@ decl_error! {
         /// Multisig can not be unlinked from an identity while it still holds POLYX
         MultiSigHasBalance,
         /// Confidential Scope claims can be added by an Identity to it-self.
-        ConfidentialScopeClaimNotAllowed
+        ConfidentialScopeClaimNotAllowed,
+        /// Addition of a new scope claim gets invalidated.
+        InvalidScopeClaim,
+        /// When scope is other than the desired scope type.
+        InvalidScopeType
     }
 }
 
```

Comment:
We need a storage migration function that validates the existing `InvestorZKProof` claims.

Reply:
I don't think anybody will have the InvestorZKProof as it is not easy to provide valid proof without having an SDK that allows a user to generate a proof with the help of a cryptography library. That is why I think it is more of a waste of time to have migrations for the invalid data.